staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -608,9 +593,40 @@ func (g *Cloud) ensureInternalInstanceGroup(name, zone string, nodes []*v1.Node)
 func (g *Cloud) ensureInternalInstanceGroups(name string, nodes []*v1.Node) ([]string, error) {
 	zonedNodes := splitNodesByZone(nodes)
 	klog.V(2).Infof("ensureInternalInstanceGroups(%v): %d nodes over %d zones in region %v", name, len(nodes), len(zonedNodes), g.region)
+
 	var igLinks []string
+	gceZonedNodes := map[string][]string{}
 	for zone, nodes := range zonedNodes {
-		igLink, err := g.ensureInternalInstanceGroup(name, zone, nodes)
+		instances, err := g.getFoundInstanceByNames(nodeNames(nodes))
```

Comment:
Can we split this into its own function and invoke it here instead of putting this code inline?

Something like: 
```
if g.externalInstanceGroupsPrefix != "" {
   return reuseInstanceGroups()
} 

// Keep the existing logic
}
```

This way it is clearer that the default behavior is not changed.

Reply:
@prameshj pushed an update that makes it conditional. PTAL